doc/reference/reference_lua/box_null.rst

```diff
@@ -0,0 +1,138 @@
+.. _box-null:
+
+-------------------------------------------------------------------------------
+                             Constant `box.NULL`
+-------------------------------------------------------------------------------
+
+There are some major problems with using a Lua **nil** value in tuples.
+For example: you can't correctly assess length of sparse array (an array with
+nullable values set to NULL) in case of using Lua **nil** value.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, nil, 1, 2, nil}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
+      - null
+      - 1
+      - 2
+    ...
+
+    tarantool> #t
+    ---
+    - 4
+    ...
+
+Provided example clearly shows a problem with sparse arrays.
+
+.. NOTE::
+
+    Trying to find length for sparse arrays in LuaJIT leads to
+    `undefined behaviour <https://www.lua.org/manual/5.2/manual.html#3.4.6>`_.
+
+To overcome this and some other issues use ``box.NULL`` constant instead of **nil**.
+
+Using box.NULL
+--------------
+
+``box.NULL`` is a value of cdata type representing a NULL pointer.
+It is similar to ``msgpack.NULL``, ``json.NULL`` and ``ffi.NULL``. So it is
+some not **nil** value, even if it is a pointer to NULL.
+
+Use ``box.NULL`` only with capitalized NULL (``box.null`` is incorrect).
+
+.. NOTE::
+
+    Technically speaking ``box.NULL`` equals to ``ffi.cast('void *', 0)``.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, box.NULL, 1, 2, box.NULL}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
+      - null
+      - 1
+      - 2
+      - null
```

Comment:
How about comments `# cdata`, The difference between this example and the previous one isn't emphasized. It's not only about t[5], but also about their types

Reply:
Also this difference isn't crucial for this illustrative examples (we are not implied to investigate all the difference, only show enough to differentiate this cases) and their types are kind of obvious, since we define them - i will still add a small note, spotting a difference, just in case.

Reply:
# cdata comment possible in 2nd example